sling-maven-plugin/src/main/java/org/apache/sling/maven/bundlesupport/JcrInstallMojo.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.sling.maven.bundlesupport.deploy.BundleDeploymentMethod;
+import org.apache.sling.maven.bundlesupport.deploy.DeployContext;
+import org.apache.sling.maven.bundlesupport.deploy.DeployMethod;
+
+import java.io.File;
+import java.util.Optional;
+
+/**
+ * Upload a non-bundle artifact to the JCR repository of a running Sling instance.
+ * This goal mirrors the behavior of the <code>install</code> goal for uploading bundles when <code>deploymentMethod</code>
+ * is set to either <code>WebDAV</code> or <code>SlingPostServlet</code>, and is skipped when <code>deploymentMethod</code>
+ * is set to <code>WebConsole</code>.
+ * For <code>WebDAV</code> the goal will use HTTP PUT to leverage the <a href="http://sling.apache.org/documentation/development/repository-based-development.html">WebDAV bundle from Sling</a>.
+ * For <code>SlingPostServlet</code>, the goal will leverage the <a href="http://sling.apache.org/documentation/bundles/manipulating-content-the-slingpostservlet-servlets-post.html">Sling POST servlet</a>
+ * instead. The chosen method depends on the parameter {@link #deploymentMethod}.
+ * <br>
+ * <p><strong>Intermediate Node Creation</strong></p>
+ * <p>
+ * For supported <code>deploymentMethod</code>s, the artifact is not directly deployed within the OSGi container,
+ * but rather is uploaded to the JCR and from there on being picked up by the
+ * <a href="https://sling.apache.org/documentation/bundles/jcr-installer-provider.html">JCR Installer Provider</a> asynchronously, which takes care 
+ * of performing any installable resource transformation supported by the OSGi container. For both supported deployment methods, intermediate nodes (i.e. inexisting parent nodes)
+ * are automatically created. The primary type of those intermediate nodes depends on the deployment method.
+ * </p>
+ * <ul>
+ * <li>
+ *  WebDAV, uses the configured collection node type, by default <code>sling:Folder</code>
+ *  (see also <a href="https://sling.apache.org/documentation/development/repository-based-development.html">WebDAV Configuration</a>)</li>
+ * <li>
+ *  SlingPostServlet, uses internally <code>ResourceResolverFactory.create(...)</code> without setting any <code>jcr:primaryType</code>.
+ *  Therefore the <code>JcrResourceProviderFactory</code> will call <code>Node.addNode(String relPath)</code> which determines a fitting 
+ *  node type automatically, depending on the parents node type definition (see <a href="https://docs.adobe.com/docs/en/spec/jsr170/javadocs/jcr-2.0/javax/jcr/Node.html#addNode(java.lang.String)">Javadoc</a>).
+ *  So in most of the cases this should be a <code>sling:Folder</code>, as this is the first allowed child node definition in <code>sling:Folder</code>.
+ *  This only may differ, if your existing parent node is not of type <code>sling:Folder</code> itself.
+ * </li>
+ * </ul>
+ *
+ * @since 2.5.0
+ */
+@Mojo(name = "jcr-install", defaultPhase = LifecyclePhase.INSTALL)
```

Comment:
I would rather tend to merge this functionality into the existing `sling:install` mojo. To be backwards compatible one should be forced into opt in via a switch like `artifactExtensions` which should by default only be `jar`. Also from a naming perspective this is not bound to JCR but supports arbitrary resources (at least for the SlingPostServlet deployment method). Obviously this requires adjusting the description of the `sling:install` mojo (to cover all types of artifacts which are configured via `artifactExtensions`). Also the parameter `bundleFileName` would need some clarification.

Reply:
I agree with @kwin. It could be interesting to use the same mechanism to install other resources, e.g. feature archives (FAR).